src/target/riscv/riscv.c

```diff
@@ -462,11 +462,76 @@ static void trigger_from_breakpoint(struct trigger *trigger,
 	trigger->unique_id = breakpoint->unique_id;
 }
 
-static int maybe_add_trigger_t1(struct target *target,
-		struct trigger *trigger, uint64_t tdata1)
+static bool match_napot_checker(struct trigger *trigger, riscv_reg_t *tdata2)
+{
+	riscv_reg_t addr = trigger->address;
+	riscv_reg_t size = trigger->length;
+	if (size > 1 &&
+		(size & (size - 1)) == 0 &&
+		(addr & (size - 1)) == 0) {
+		if (tdata2)
+			*tdata2 = addr | ((size - 1) >> 1);
+		return true;
+	}
+	return false;
+}
+
+static int find_trigger(struct target *target, int type, bool chained, int *idx)
 {
 	RISCV_INFO(r);
 
+	int found;
+	int cnt = 0;
+	int num = chained ? 2 : 1;
```

Comment:
Please give the variables a more descriptive names, for example:

```suggestion
	unsigned int num_found = 0;
	unsigned int num_required = chained ? 2 : 1;
```